pydantic/_internal/_std_types_schema.py

```diff
@@ -67,58 +67,65 @@ def get_enum_core_schema(enum_type: type[Enum], config: ConfigDict) -> CoreSchem
     js_updates = {'title': enum_type.__name__, 'description': description}
     js_updates = {k: v for k, v in js_updates.items() if v is not None}
 
-    def get_json_schema(_, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
-        json_schema = handler(core_schema.literal_schema([x.value for x in cases], ref=enum_ref))
-        original_schema = handler.resolve_ref_schema(json_schema)
-        update_json_schema(original_schema, js_updates)
-        return json_schema
-
-    if not cases:
-        # Use an isinstance check for enums with no cases.
-        # The most important use case for this is creating TypeVar bounds for generics that should
-        # be restricted to enums. This is more consistent than it might seem at first, since you can only
-        # subclass enum.Enum (or subclasses of enum.Enum) if all parent classes have no cases.
-        # We use the get_json_schema function when an Enum subclass has been declared with no cases
-        # so that we can still generate a valid json schema.
-        return core_schema.is_instance_schema(enum_type, metadata={'pydantic_js_functions': [get_json_schema]})
-
-    if len(cases) == 1:
-        expected = repr(cases[0].value)
-    else:
-        expected = ', '.join([repr(case.value) for case in cases[:-1]]) + f' or {cases[-1].value!r}'
-
-    def to_enum(input_value: Any, /) -> Enum:
-        try:
-            return enum_type(input_value)
-        except ValueError:
-            raise PydanticCustomError('enum', 'Input should be {expected}', {'expected': expected})
-
+    sub_type: typing_extensions.Literal['str', 'int', 'float'] | None = None
     if issubclass(enum_type, int):
-        # this handles `IntEnum`, and also `Foobar(int, Enum)`
-        js_updates['type'] = 'integer'
-        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema())
+        sub_type = 'int'
+        value_ser_type: core_schema.SerSchema = core_schema.simple_ser_schema('int')
     elif issubclass(enum_type, str):
         # this handles `StrEnum` (3.11 only), and also `Foobar(str, Enum)`
-        js_updates['type'] = 'string'
-        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.str_schema())
+        sub_type = 'str'
+        value_ser_type = core_schema.simple_ser_schema('str')
     elif issubclass(enum_type, float):
-        js_updates['type'] = 'numeric'
-        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.float_schema())
+        sub_type = 'float'
+        value_ser_type = core_schema.simple_ser_schema('float')
     else:
```

Comment:
Optional change, could do something like

```py
allowed_enum_types = [int, str, float]

for type_ in allowed_enum_types:
    if issubclass(enum_type, type_):
        sub_type = type_.__name__
        value_ser_type = core_schema.simple_ser_schema(sub_type)
        break
else:
    value_ser_type = core_schema.plain_serializer_function_ser_schema(lambda x: x)
```

Re the `lambda x: x` question for an any schema - I'm not sure. @davidhewitt, `any` ideas? Haha

Reply:
I'm guessing my above suggestion might be a bit less friendly with type checkers, but idk

Reply:
RE the above, could always do `allowed_enum_types = [(int, 'int'), ...]` etc to avoid the `__name__` bit if type checkers don't like it. No strong opinion which form is better.

In `SerializeAsAny` it is:

 ```
            core_schema.wrap_serializer_function_ser_schema(
                lambda x, h: h(x), schema=core_schema.any_schema()
            )
```

But really we should be able to avoid going to Python for this case, so I think we should write a todo for us to add support to core.

Reply:
https://github.com/pydantic/pydantic-core/issues/1245, marking as resolved.